Massive remove of word Please from whole code repository

igor2x opened the pull request:
Remove word "Please" from whole code repository. Once and final simplification.

dskrvk commented on src/core/core.rc2:
```diff
@@ -91,7 +91,7 @@ BEGIN
   IDSC_IMPORTCONFLICTSX1  "Entry with Group=""%ls"""
   IDSC_IMPORTCONFLICTSX2  "Entry with no group"
-  IDSC_IMPORTABORTED      "Import aborted. Please see report."
+  IDSC_IMPORTABORTED      "Import aborted. See report."
```
Maybe keep it here? My thinking is, the word here softens the user frustration from the operation failing.

dskrvk commented on src/core/core.rc2:
```diff
@@ -173,8 +173,8 @@ BEGIN
-  IDSC_FILE_UNREADABLE    "Cannot read file. Please check file permissions."
-  IDSC_FILE_TRUNCATED     "File is truncated or otherwise corrupt. Please use backup database."
+  IDSC_FILE_UNREADABLE    "Cannot read file. Check file permissions."
+  IDSC_FILE_TRUNCATED     "File is truncated or otherwise corrupt. Use backup database."
```
Same for these two lines

dskrvk commented on src/ui/Windows/DboxMain.cpp:
```diff
@@ -3491,7 +3491,7 @@ int DboxMain::OnUpdateMenuToolbar(const UINT nID)
   if (it == m_MapUICommandTable.end()) {
     // Don't have it - allow by default
-    pws_os::Trace(L"Menu resource ID: %d not found in m_UICommandTable. Please investigate and correct.\n", nID);
+    pws_os::Trace(L"Menu resource ID: %d not found in m_UICommandTable. Investigate and correct.\n", nID);
```
Same

dskrvk commented on src/ui/cli/safeutils.cpp:
```diff
@@ -97,11 +97,11 @@ StringX GetNewPassphrase()
         }
         if (passphrase[0].length() == 0) {
-            wcerr << "Invalid passphrase. Please try again." << endl;
+            wcerr << "Invalid passphrase. Try again." << endl;
```
Same for both lines in this file

ronys replied:
Agree.
"Please try again."

dskrvk commented on src/ui/wxWidgets/ExternalKeyboardButton.cpp:
```diff
@@ -68,15 +68,15 @@ void ExternalKeyboardButton::HandleCommandEvent(wxCommandEvent& evt)
 
   if (!pws_os::ProgramExists(tostdstring(command))) {
-    wxMessageBox(_("Could not launch xvkbd.  Please make sure it's installed and in your PATH"), 
+    wxMessageBox(_("Could not launch xvkbd.  Make sure it's installed and in your PATH"), 
```
Same about the three lines changed in this file

ronys replied:
agree.
"Please make sure it's ..."

dskrvk commented on src/ui/wxWidgets/MenuFileHandlers.cpp:
```diff
@@ -875,7 +875,7 @@ void PasswordSafeFrame::DoImportText(wxString filename)
     wxMessageBox(wxString() << _("The database:") << wxT("\n\n") << m_core.GetCurFile() << wxT("\n\n")
                             << _("has duplicate entries with the same group/title/user combination.")
-                            << _("  Please fix by validating database."),
+                            << _("  Fix by validating database."),
```
Side note: we shouldn’t be conctenating translatable strings like this because other languages have different word orders. We should be using format placeholders instead.

ronys replied:
Agree, but that's another PR...

dskrvk commented on src/ui/wxWidgets/OpenFilePickerValidator.cpp:
```diff
@@ -60,7 +60,7 @@ bool OpenFilePickerValidator::Validate(wxWindow * parent) {
       //path is blank on Linux/gtk. May be its not so on other platforms
       wxMessageBox(wxString() << _("Selected file doesn't exist.") << wxT("\n\n") << path,
-                              _("Please select a valid file"), wxOK | wxICON_EXCLAMATION, parent);
+                              _("Select a valid file"), wxOK | wxICON_EXCLAMATION, parent);
```
Keep the old text?

ronys replied:
Please :-)

dskrvk commented on src/ui/wxWidgets/PWFiltersDateDlg.cpp:
```diff
@@ -589,7 +589,7 @@ bool pwFiltersDateDlg::IsValid(bool showMessage) const {
     if ((fdate1 > wxDateTime::Now()) || (isRuleSelected(idx, PWSMatch::MR_BETWEEN) && (fdate2 > wxDateTime::Now()))) {
       if (showMessage) {
-        wxMessageBox(_("Please correct date values."), _("A future date is not valid for this field"), wxOK|wxICON_ERROR);
+        wxMessageBox(_("Correct date values."), _("A future date is not valid for this field"), wxOK|wxICON_ERROR);
```
I would keep the old messages in this file

ronys replied:
Agree.

dskrvk commented on src/ui/wxWidgets/PWFiltersIntegerDlg.cpp:
```diff
@@ -500,26 +500,26 @@ bool pwFiltersIntegerDlg::IsValid(bool showMessage) const
   else if(isRuleSelected(idx, PWSMatch::MR_GT) && (fnum1 == m_max)) {
     if (showMessage) {
-      wxMessageBox(_("Please correct numeric values."), _("Number is set to the maximum value. 'Greater than' is not allowed."), wxOK|wxICON_ERROR);
+      wxMessageBox(_("Correct numeric values."), _("Number is set to the maximum value. 'Greater than' is not allowed."), wxOK|wxICON_ERROR);
```
Same in this file

ronys replied:
ditto

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -302,32 +302,32 @@ BEGIN
+  IDS_NAMETEXTFILE        "Name the plaintext file"
+  IDS_NAMEXMLFILE         "Name the XML file"
+  IDS_NAMEDBFILE          "Name the database file"
   IDS_BADPASSKEY          "Passkey incorrect"
```
"Incorrect master password"

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -302,32 +302,32 @@ BEGIN
+  IDS_NAMEDBFILE          "Name the database file"
   IDS_BADPASSKEY          "Passkey incorrect"
-  IDS_MISSINGPASSKEY      "Passkey needed to save. Please unlock database and try again"
+  IDS_MISSINGPASSKEY      "Passkey needed to save. Unlock database and try again"
```
"Master password needed to save. Please unlock database and try again"

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -458,11 +458,11 @@ END
   IDS_UNTITLEDRESTORE     "Password Safe - <Untitled Restored Backup>"
   IDS_VALIDATE_OK         "Database validated - no problems found."
-  IDS_VALIDATE_ISSUES     "Some consistency issues were found while opening the database.\n\nYou should save this database before proceeding. Please view report for details."
+  IDS_VALIDATE_ISSUES     "Some consistency issues were found while opening the database.\n\nYou should save this database before proceeding. View report for details."
```
"More details may be found in the report."

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -590,16 +590,16 @@ BEGIN
 STRINGTABLE
 BEGIN
-  IDS_DBIDINUSE             "This index is currently in use. Please select another."
+  IDS_DBIDINUSE             "This index is currently in use. Select another."
```
"Please select another."

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -646,12 +646,12 @@ BEGIN
-  IDS_FILTERINCOMPLETE      "Row %d is incomplete. Please set both Field and Criteria."
+  IDS_NOSTRING              "Specify text"
+  IDS_NOMEDIATYPE           "Select media type"
+  IDS_FILTERINCOMPLETE      "Row %d is incomplete. Set both Field and Criteria."
```
"Both field and Criteria must be set."

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -726,9 +726,9 @@ BEGIN
   IDS_REPLACEFILTER         "Do you wish to replace it?"
   IDS_FILTEREXISTS          "This filter already exists"
-  IDS_CONFIRMFILTERDELETE   "Please confirm delete of the %s filter %s?"
+  IDS_CONFIRMFILTERDELETE   "Confirm delete of the %s filter %s?"
```
"Confirm deletion of the %s filter %s?"

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -764,7 +764,7 @@ END
 BEGIN
   IDS_CANTLOAD_AUTOTYPEDLL "Failed to load AutoType support DLL (wrong version?).\n\nAutoType for Launch of a Browser or Running a command is disabled."
-  IDS_OSK_VERSION_MISMATCH "Version mismatch with On Screen Keyboard DLL. Please install version that came with this version of Password Safe."
+  IDS_OSK_VERSION_MISMATCH "Version mismatch with On Screen Keyboard DLL. Install version that came with this version of Password Safe."
```
"Please install the dll that came with this version of Password Safe."

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -1017,10 +1017,10 @@ END
   IDS_YUBI_TIMEOUT            "No response from YubiKey"
   IDS_YUBI_UNINITIALIZED      "The YubiKey appears uninitialized. Initialize it?"
-  IDS_YUBI_INIT_FAILED        "Failed to initialize YubiKey. Please try again."
+  IDS_YUBI_INIT_FAILED        "Failed to initialize YubiKey. Try again."
```
"Please try again."

ronys commented on src/ui/cli/safeutils.cpp:
```diff
@@ -97,11 +97,11 @@ StringX GetNewPassphrase()
 
         if (passphrase[0] != passphrase[1]) {
-            wcerr << "The two passphrases do not match. Please try again." << endl;
+            wcerr << "The two passphrases do not match. Try again." << endl;
```
"Please try again."

ronys commented on src/ui/wxWidgets/CompareDlg.cpp:
```diff
@@ -814,7 +814,7 @@ void CompareDlg::DoSyncItemsWithCurrentDB(int menuId, ContextMenuData menuContex
   if (!m_currentCore->GetUniqueGTUValidated() && !m_currentCore->InitialiseGTU(setGTU)) {
     // Database is not unique to start with - tell user to validate it first
-    wxMessageBox(wxString::Format(_("The database:\n\n%ls\n\nhas duplicate entries with the same group/title/user combination. Please fix by validating database.").c_str(), m_currentCore->GetCurFile().c_str()),
+    wxMessageBox(wxString::Format(_("The database:\n\n%ls\n\nhas duplicate entries with the same group/title/user combination. Fix by validating database.").c_str(), m_currentCore->GetCurFile().c_str()),
```
"You can fix this by validating the database."

ronys commented on src/ui/wxWidgets/ExternalKeyboardButton.cpp:
```diff
@@ -68,15 +68,15 @@ void ExternalKeyboardButton::HandleCommandEvent(wxCommandEvent& evt)
   {
     case 0:
-      wxMessageBox(_("Could not launch xvkbd.  Please make sure it's in your PATH"), 
+      wxMessageBox(_("Could not launch xvkbd.  Make sure it's in your PATH"), 
```
"Please make sure..."

ronys commented on src/ui/wxWidgets/ExternalKeyboardButton.cpp:
```diff
@@ -95,7 +95,7 @@ void ExternalKeyboardButton::HandleCommandEvent(wxCommandEvent& evt)
 
   if ( wxExecute(command, wxEXEC_ASYNC, nullptr) > 0) {
-    wxMessageBox(_("Please, enable the Accessibility Keyboard in System Settings; the on-screen keyboard should then appear."), "", wxOK | wxICON_INFORMATION);
+    wxMessageBox(_("Enable the Accessibility Keyboard in System Settings; the on-screen keyboard should then appear."), "", wxOK | wxICON_INFORMATION);
```
"Please enable the..."

ronys commented on src/ui/wxWidgets/MenuFileHandlers.cpp:
```diff
@@ -993,7 +993,7 @@ void PasswordSafeFrame::DoImportXML(wxString filename)
   if (!m_core.GetUniqueGTUValidated() && !m_core.InitialiseGTU(setGTU)) {
     // Database is not unique to start with - tell user to validate it first
-    wxMessageBox(wxString::Format( _("The database:\n\n%ls\n\nhas duplicate entries with the same group/title/user combination. Please fix by validating database."),
+    wxMessageBox(wxString::Format( _("The database:\n\n%ls\n\nhas duplicate entries with the same group/title/user combination. Fix by validating database."),
```
"You can fix this by validating the database."

ronys commented on src/ui/wxWidgets/MenuFileHandlers.cpp:
```diff
@@ -1004,7 +1004,7 @@ void PasswordSafeFrame::DoImportXML(wxString filename)
   if (!XSDFilename.FileExists()) {
     wxString filepath(XSDFilename.GetFullPath());
-    wxMessageBox(wxString::Format(_("Can't find XML Schema Definition file (%ls) in your PasswordSafe Application Directory.\nPlease copy it from your installation file, or re-install PasswordSafe."), filepath.c_str()),
+    wxMessageBox(wxString::Format(_("Can't find XML Schema Definition file (%ls) in your PasswordSafe Application Directory.\nCopy it from your installation file, or re-install PasswordSafe."), filepath.c_str()),
```
"Please copy it..."

ronys commented on src/ui/wxWidgets/PWSafeApp.cpp:
```diff
@@ -930,7 +930,7 @@ void PWSafeApp::OnHelp(wxCommandEvent& evt)
     else {
 #ifdef __WXDEBUG__
-      msg << _("Please inform the developers.");
+      msg << _("Inform the developers.");
```
Definitely. "Inform the developers" sounds like the user is working for them...

ronys commented on src/ui/wxWidgets/SafeCombinationChangeDlg.cpp:
```diff
@@ -368,7 +368,7 @@ void SafeCombinationChangeDlg::OnYubibtn2Click(wxCommandEvent& WXUNUSED(event))
     if (m_oldpasswd.empty() && m_oldresponse.empty()) {
       m_yubiStatusCtrl->SetForegroundColour(*wxRED);
-      m_yubiStatusCtrl->SetLabel(_("Please confirm existing combination"));
+      m_yubiStatusCtrl->SetLabel(_("Confirm existing combination"));
```
"Confirm existing master password"

ronys commented on src/ui/wxWidgets/SyncWizard.cpp:
```diff
@@ -670,7 +670,7 @@ bool SyncStatusPage::DbHasNoDuplicates(PWScore* core)
     // Database is not unique to start with - tell user to validate it first
     SetSyncSummary(_("Synchronization failed"));
-    SetProgressText(wxString::Format(_("The database:\n\n%ls\n\nhas duplicate entries with the same group/title/user combination. Please fix by validating database."), core->GetCurFile().c_str()));
+    SetProgressText(wxString::Format(_("The database:\n\n%ls\n\nhas duplicate entries with the same group/title/user combination. Fix by validating database."), core->GetCurFile().c_str()));
```
"You can fix this by validating the database."

ronys reviewed (changes requested):
I agree with @dskrvk that a total purge of "Please" is too strong.
Removing it from "happy flow" windows makes sense, but for errors it is more "humane", for lack of a better word.
Also in some places the text becomes to imperative. I've tweaked the text a bit to make it more friendly without resorting to "please".

seamusdemora commented:
Silly question, but may I ask what is driving this *quest* to remove the word "please" from the `pwsafe` repository?

igor2x commented:
@ronys, I don't want this commit to become long lasting "major project", so I did requested changes and reverted and/or changed strings as requested. From my perspective, at least some of the "Please" words are removed as redundant. Tiny improvement, that make me happy.

In general, in my humble opinion "Please" word all over the program is unusual design. In most cases instead of "Please" word, sentence should be rewritten (but this out of scope for this PR). In my humble opinion "Please" word should only be used if we (as a program) are causing user to feel inconvenience, like "This will take some minutes. Please wait." In this case we as a program are causing inconvenience to user. Another sample is, "USB key not detected. Please re-insert it." We as a program should detect the key, but we failed, and as the consequence, user must go and unplug and plug it again, so we caused user inconvenience. On the other hand, if user did something wrong, like typing in wrong master password, then this is not the inconvenience we as a program caused, this is user fault and in this case there should be no "Please" word.

@seamusdemora, simplification.

ronys commented on src/ui/Windows/res/PasswordSafe3.rc2:
```diff
@@ -646,12 +646,12 @@ BEGIN
-  IDS_FILTERINCOMPLETE      "Row %d is incomplete. Please set both Field and Criteria."
+  IDS_NOSTRING              "Specify text"
+  IDS_NOMEDIATYPE           "Select media type"
+  IDS_FILTERINCOMPLETE      "Row %d is incomplete. Both field and Criteria must be set."
```
"Field" should be capitalized.

ronys commented:
I agree that "Please" is currently overused, but many review comments suggest alternate phrasing, or keep it for error handling. Can you go over them again?

dskrvk commented:
> On the other hand if we would like to preserve word "Please" to reduce users frustration, then we most probably have bigger problem: non-optimal engineering of dialogs. In many cases a lot of confirmations can be reduced to minimal. For example, if we delete an entry, we can do it asking user like "Please reconsider the delete entry" and we force decision to be made about deleting, that actually user already did, when he or she decided to delete an entry. But in my humble opinion this is not the most friendly design. Why? Because in 99% of the cases, user wants to delete the entry and in 1% or less user pointed to wrong entry or selected the wrong action from menu. In this rare cases the design could be like this: We delete entry immediately and in status bar we write what entry program has deleted and offer the undo link in status bar. What we did is in 99% of the cases we spared user dialog box decision and click on confirmation button, and in 1% we offered easy to undo the task. We reduce frustration with better dialog design. But dialog design improvements exceeds this easy to understand PR, which main goal is simplification of labels in dialogs. Other less-frustrating dialog improvements should be made in separate PRs. Idea, don't do one huge improvement in one PR, but instead make thousands small PR improvements that combined/summarized will count as a major leap.

FWIW, I agree that the approach “don’t ask for confirmation, offer to undo” is more user-friendly,  but implementing it would be a bigger task. In the meantime, I think we should keep the messages polite where makes sense. When suggesting my changes, my logic was “if it’s our fault, let’s keep it polite; if it’s the user’s, we can be more terse”.

igor2x commented:
@rony, in yesterday commits I have already reverted a lot of Please words and added some suggested changes.
Few minutes ago  I have added additional one changed suggested from your yesterday review.
